scripts/protect.sh

```diff
@@ -0,0 +1,234 @@
+      | jq '.allwikis | keys[]' -r
+  )
+  # Don't get rate limited
+  sleep 4
+
+}
+
+checkForLocalVersion() {
+  if [[ $2 == "commons" ]]; then
+    hasNoLocalVersion=false
+  elif [[ $allLuaFiles == *"lua/wikis/${2}/${1}.lua"* ]] || [[ $filesToProtect == *"lua/wikis/${2}/${1}.lua"* ]]; then
+    hasNoLocalVersion=false
+  else
+    hasNoLocalVersion=true
+  fi
+}
+
+protectPage() {
+  protectOptions=$3
+  wiki=$2
+  page="Module:${1}"
+  echo "...wiki = $wiki"
+  echo "...page = $page"
+  wikiApiUrl="${WIKI_BASE_URL}/${wiki}/api.php"
+  ckf="cookie_${wiki}.ck"
+
+  if [[ ${loggedin[${wiki}]} != 1 ]]; then
+    # Login
+    echo "...logging in on \"${wiki}\""
+    loginToken=$(
+      curl \
+        -s \
+        -b "$ckf" \
+        -c "$ckf" \
+        -d "format=json&action=query&meta=tokens&type=login" \
+        -H "User-Agent: ${userAgent}" \
+        -H 'Accept-Encoding: gzip' \
+        -X POST "$wikiApiUrl" \
+        | gunzip \
+        | jq ".query.tokens.logintoken" -r
+    )
+    curl \
+      -s \
+      -b "$ckf" \
+      -c "$ckf" \
+      --data-urlencode "lgname=${WIKI_USER}" \
+      --data-urlencode "lgpassword=${WIKI_PASSWORD}" \
+      --data-urlencode "lgtoken=${loginToken}" \
+      -H "User-Agent: ${userAgent}" \
+      -H 'Accept-Encoding: gzip' \
+      -X POST "${wikiApiUrl}?format=json&action=login" \
+      | gunzip \
+      > /dev/null
+    loggedin[$wiki]=1
+    # Don't get rate limited
+    sleep 4
+  fi
+
+  # Protect Page
+  protectToken=$(
+    curl \
+      -s \
+      -b "$ckf" \
+      -c "$ckf" \
+      -d "format=json&action=query&meta=tokens" \
+      -H "User-Agent: ${userAgent}" \
+      -H 'Accept-Encoding: gzip' \
+      -X POST "$wikiApiUrl" \
+      | gunzip \
+      | jq ".query.tokens.csrftoken" -r
+  )
+  rawProtectResult=$(
+    curl \
+      -s \
+      -b "$ckf" \
+      -c "$ckf" \
+      --data-urlencode "title=${page}" \
+      --data-urlencode "protections=${protectOptions}" \
+      --data-urlencode "reason=Git maintained" \
+      --data-urlencode "expiry=infinite" \
+      --data-urlencode "bot=true" \
+      --data-urlencode "token=${protectToken}" \
+      -H "User-Agent: ${userAgent}" \
+      -H 'Accept-Encoding: gzip' \
+      -X POST "${wikiApiUrl}?format=json&action=protect" \
+      | gunzip
+  )
+  # Don't get rate limited
+  sleep 4
+}
+
+protectExistingPage() {
+  protectPage $1 $2 "edit=allow-only-sysop|move=allow-only-sysop"
+  result=$(echo "$rawProtectResult" | jq ".protect.protections.[].edit" -r)
+  if [[ $result != *"allow-only-sysop"* ]]; then
+    echo "::warning::could not protect $1 on $2 against editing"
+    protectErrors+=("$1 on $2")
+  fi
+}
+
+protectNonExistingPage() {
+  protectPage $1 $2 "create=allow-only-sysop"
+  result=$(echo "$rawProtectResult" | jq ".protect.protections.[].create" -r)
+  if [[ $result != *"allow-only-sysop"* ]]; then
+    echo "::warning::could not protect $1 on $2 against creation"
+    protectErrors+=("$1 on $2")
+  fi
+}
```

Comment:
Is this required to be separate, i.e. fails if it isn't?
What is the protection status of a page that has `create=allow-only-sysop`, and is then created? Can anyone edit afterwards?

Reply:
> Is this required to be separate, i.e. fails if it isn't?

need to check different paths in the json (edit vs create)
could probably move it into the `protectPage` by passing an additional param (`create`/`edit`)

> What is the protection status of a page that has create=allow-only-sysop, and is then created? Can anyone edit afterwards?

a page that is protected against creation can only be created by an admin
admins should know what they are doing, especially as the protection message states "Git maintained" (see e.g. [here(https://liquipedia.net/wildcard/index.php?title=Module:Infobox/Weapon&action=edit))

if an admin creates the page and opens a PR to add it to the git the PR will "create" the page and lock it down to admin with that "creation" (merge)

Reply:
[dcae3bb](https://github.com/Liquipedia/Lua-Modules/pull/5705/commits/dcae3bb8dee6851ac9c478be673a155c9cac30bc)